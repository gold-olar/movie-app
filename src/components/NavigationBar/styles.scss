.logo {
  font-size: 150%;
  color: #fff;
  font-weight: 900;
}
.logo-red {
  color: #db0000;
}
.logo-link {
  text-decoration: none !important;
}
.nav-link{
  font-weight: 900;
  text-transform: uppercase;
  letter-spacing: 1.5px;
}
.active{
  color: #db0000!important;
}
