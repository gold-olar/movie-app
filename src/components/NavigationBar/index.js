import React, { useState } from "react";
import { Collapse, Navbar, NavbarToggler, Nav, NavItem } from "reactstrap";
import { NavLink } from "react-router-dom";
import "./styles.scss";

const NavigationBar = () => {
  const [isOpen, setIsOpen] = useState(false);

  const toggle = () => setIsOpen(!isOpen);

  return (
    <div>
      <Navbar color="transparent" dark expand="md">
        {" "}
        <NavLink className="nav-brand logo-link" to="/">
          {" "}
          <span className="logo">
            {" "}
            Movie<span className="logo-red">App</span>{" "}
          </span>{" "}
        </NavLink>{" "}
        <NavbarToggler onClick={toggle} />
        <Collapse isOpen={isOpen} navbar>
          <Nav className="ml-auto" navbar>
            <NavItem>
              <NavLink className="nav-link" to="/series">
                Series
              </NavLink>
            </NavItem>
            <NavItem>
              <NavLink className="nav-link" to="/movies">
                Movies
              </NavLink>
            </NavItem>
          </Nav>
        </Collapse>
      </Navbar>
    </div>
  );
};

export default NavigationBar;
