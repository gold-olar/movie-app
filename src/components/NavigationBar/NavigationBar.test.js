import React from "react";
import { BrowserRouter } from "react-router-dom";
import { unmountComponentAtNode } from "react-dom";
import ReactDOM from "react-dom";
import NavigationBar from "./index";
import renderer from "react-test-renderer";

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
});
const testComponent = (
  <BrowserRouter>
    <NavigationBar type="series" description="any string" path="/path" />{" "}
  </BrowserRouter>
);
describe("Navigation Bar", () => {
  test("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(testComponent, div);
  });

  test("should match snapshot", () => {
    const component = renderer.create(testComponent);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
