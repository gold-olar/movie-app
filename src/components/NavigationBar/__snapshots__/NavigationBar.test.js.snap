// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Bar should match snapshot 1`] = `
Array [
  <div>
    <nav
      className="navbar navbar-expand-md navbar-dark bg-transparent"
    >
       
      <a
        aria-current="page"
        className="nav-brand logo-link active"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
         
        <span
          className="logo"
        >
           
          Movie
          <span
            className="logo-red"
          >
            App
          </span>
           
        </span>
         
      </a>
       
      <button
        aria-label="Toggle navigation"
        className="navbar-toggler"
        onClick={[Function]}
        type="button"
      >
        <span
          className="navbar-toggler-icon"
        />
      </button>
      <div
        className="collapse navbar-collapse"
        style={Object {}}
      >
        <ul
          className="ml-auto navbar-nav"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/series"
              onClick={[Function]}
            >
              Series
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-link"
              href="/movies"
              onClick={[Function]}
            >
              Movies
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>,
  " ",
]
`;
